### PR TITLE
Add a note about `airbyte_hidden` spec fields in the connector reference docs

### PR DESCRIPTION
**Pull request by sherifnada**
Realized we didn't describe this field in the connector spec reference doc, so adding it

### CODE REVIEW

File: docs/connector-development/connector-specification-reference.md [hunk 1]
```diff
@@ -77,6 +77,32 @@ By default, string inputs in the UI can lose their linebreaks. In order to accep
...
+
+"first_name": {
+  "type": "string",
+  "title": "First Name"
+}, 
+"secret_name": {
+  "type": "string", 
+  "title": "You can't see me!!!",
+  "airbyte_hidden": true
+}
```
Comment (evantahler) [inline docs/connector-development/connector-specification-reference.md:96]: The picture below seems to imply 3 fields: name (required), first_name, and secret_name.  This should probably match that.

File: docs/connector-development/connector-specification-reference.md [hunk 2]
```diff
@@ -77,6 +77,32 @@ By default, string inputs in the UI can lose their linebreaks. In order to accep
...
+
+}
+```
+
+Results in the following form: 
+
+<img width="586" alt="Screen Shot 2022-04-21 at 2 40 33 PM" src="https://user-images.githubusercontent.com/6246757/164556606-48a53725-f075-4c3f-88fb-68920e78cc32.png">
```
Comment (evantahler) [inline docs/connector-development/connector-specification-reference.md:103]: [comment] I'd suggest checking in these images to the repo directly so they can be updated and found quickly.  But... if none of the other images are checked in, 🤷

**Review by evantahler — CHANGES_REQUESTED** [review]

File: docs/connector-development/connector-specification-reference.md (hunk 1 above)
Comment (sherifnada) [reply docs/connector-development/connector-specification-reference.md:96]: Woop, good call. That's just the `Name` field which is present on every connector's configuration page, it's not part of any connector's spec. Will crop it out ✂️

Comment (sherifnada) [reply docs/connector-development/connector-specification-reference.md]: Done

**Review by evantahler — APPROVED** [review]
🥳

File: docs/connector-development/connector-specification-reference.md (hunk 2 above)
Comment (sherifnada) [reply docs/connector-development/connector-specification-reference.md]: done!

**Pull request merged**